R/transform_point_model_out.R

```diff
@@ -0,0 +1,75 @@
+#' Transform mean/median model output into a forecast object
+#'
+#' @param model_out_tbl Forecast model output tibble
+#' @param target_data Observed 'ground truth' data to be compared against forecasts
+#' @param output_type Forecast output type: "mean" or "median"
+#'
+#' @return forecast_point
+transform_point_model_out <- function(model_out_tbl, target_data, output_type) {
+  if ((!inherits(output_type, "character")) || (!output_type %in% c("mean", "median"))) {
+    cli::cli_abort(
+      "invalid 'output_type': {.val {output_type}} Must be 'mean' or 'median'"
+    )
+  }
+
+  # check that: model_out_tbl contains columns: model_id, output_type, output_type_id, value
+  req_cols <- c("model_id", "output_type", "output_type_id", "value")
+  if (!all(req_cols %in% colnames(model_out_tbl))) {
+    cli::cli_abort(
+      "model_out_tbl does not contain required columns: model_id, output_type, output_type_id, value:
+      {.val {colnames(model_out_tbl)}}"
+    )
+  }
+
+  model_out_cols <- colnames(model_out_tbl)
+  # To-do: hubUtils::std_colnames (non-task id cols)
+  non_task_cols <- c(
+    "model_id", "output_type", "output_type_id",
+    "value", "model_abbr", "team_abbr"
+  )
+  task_id_cols <- model_out_cols[!model_out_cols %in% non_task_cols]
+
+  # check that model_out_tbl and target_data have compatible columns
+  if (length(task_id_cols[task_id_cols %in% colnames(target_data)]) == 0) {
+    cli::cli_abort(
+      "model_out_tbl and target_data do not have compatible columns"
+    )
+  }
+
+  if (!c("observation") %in% colnames(target_data)) {
+    cli::cli_abort(
+      "target_data does not have observation column"
+    )
+  }
+
+  if (!inherits(model_out_tbl, "model_out_tbl")) {
+    model_out_tbl <- hubUtils::as_model_out_tbl(model_out_tbl)
+  }
+
+  type <- output_type
+
+  model_out_tbl <- model_out_tbl |>
+    dplyr::filter(output_type == type) |>
```

Comment:
I am now seeing other `dplyr` functions, so maybe it should be kept as an import, although I do think using the base R pipe is a fine choice most of the time.

Reply:
pinging @annakrystalli to weigh in on the dependencies question.

Reply:
I agree that given we are using `dplyr` functions with no simple base replacements (e.g. joins), it should be ok to use it as a dependency. Agree though that were it's easy to use base for something (e.g. selecting columns, simple filtering) to do so, even when taking `dplyr` as a dependency as the package has an annoying habit of changing API more often than we would like.

Re the pipe, when we started with the hubverse, we had made a decision to use the `%>%` pipe so as not to force a dependency on R >= 4.1, hence it is used throughout `hubData`, `hubUtils`, `hubAdmin` and `hubValidations` and even re-exported as part of them.   However, that was a year and a half ago so it's perhaps safe to start using `|>` instead.

As an aside, it is possible to import the `%>%`  from the much more lightweight `magrittr` package, which is where the tidyverse pipe originated. This is the approach we've used in packaged that don't require `dplyr`.